### PR TITLE
subscriber: make it Subscriber.Topic(name).Subscribe(...)

### PR DESCRIPTION
**Pull request by mxmCherry**
I haven't done any shortcut constructor like `ResolveSubTopic(ctx, "impl://...?topic=foo")`.

### CODE REVIEW

File: subscribe.go [hunk 1]
```diff
@@ -169,21 +175,48 @@ func NewInMemSubscriber(messagesByTopic map[string][]SubMessage) *InMemSubscribe
...
+	return NewInMemSubTopic(s.msgs[topic])
+}
+
+// Close forgets seeded messages.
+func (s *InMemSubscriber) Close() error {
+	s.mu.Lock()
+	s.msgs = nil
+	s.mu.Unlock()
+	return nil
+}
```
Comment (mxmCherry) [inline subscribe.go:196]: @dim I did not bother with remembering topics in InMemSubscriber (to clean them up on close).

Btw, I'd prefer to have not in-mem mocks, but smth channel-backed:

```go
pub, sub := NewChannel( /* buffer size */ 0 )
_ = pub.Topic("foo").Publish(...)
_ = sub.Topic("foo").Subscribe(...)
```

This is nice to check background subscriptions (like, updates subscription, that can be ran in background by lib user).

Comment (dim) [reply subscribe.go:196]: that for mem only?

File: file/file.go [hunk 2]
```diff
@@ -132,8 +132,30 @@ func NewSubscriber(root string) bps.Subscriber {
...
+	return NewSubTopic(filepath.Join(s.root, name))
+}
+
+func (s *fileSub) Close() error {
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+
+type subTopic struct {
```
Comment (dim) [inline file/file.go:145]: why not just `type subTopic string`

File: file/file.go [hunk 3]
```diff
@@ -132,8 +132,30 @@ func NewSubscriber(root string) bps.Subscriber {
...
+
+// NewSubTopic inits a subscriber topic for given file name.
+// Useful for testing.
+func NewSubTopic(filename string) bps.SubTopic {
+	return &subTopic{
+		filename: filename,
+	}
+}
+
+func (t *subTopic) Subscribe(ctx context.Context, handler bps.Handler, _ ...bps.SubOption) error {
```
Comment (dim) [inline file/file.go:157]: much more light-weight than a pointer to a struct, `func (t subTopic) Subscribe(...)`

File: publish.go [hunk 4]
```diff
@@ -59,18 +59,21 @@ func RegisterPublisher(scheme string, factory PublisherFactory) {
...
 	Attributes map[string]string `json:"attributes,omitempty"`
 }
 
-// Topic is a publisher handle to a topic.
-type Topic interface {
+// PubTopic is a publisher handle to a topic.
+type PubTopic interface {
 	// Publish publishes a message to the topic.
 	Publish(context.Context, *PubMessage) error
 	// PublishBatch publishes a batch of messages to the topic.
```
Comment (dim) [inline publish.go:79]: separate PR, but I think we should remove this for now, we can figure out something else later if we need it

File: subscribe.go (hunk 1 above)
Comment (mxmCherry) [reply subscribe.go:196]: @dim I don't really get the question. Channel-based pub-sub - for testing, mem only - if you meant this.

File: file/file.go (hunk 2 above)
Comment (mxmCherry) [reply file/file.go:145]: `type SubTopic string` then to get rid of constructor

File: file/file.go [hunk 5]
```diff
@@ -142,20 +142,13 @@ func (s *fileSub) Close() error {
...
-
-// NewSubTopic inits a subscriber topic for given file name.
+// SubTopic is an adapter for filename, that behaves like bps.SubTopic.
 // Useful for testing.
-func NewSubTopic(filename string) bps.SubTopic {
-	return &subTopic{
-		filename: filename,
-	}
-}
+type SubTopic string
```
Comment (dim) [inline file/file.go:147]: wait. does it need to be public?

Comment (mxmCherry) [reply file/file.go:147]: I think so - it's much easier to test if it's public if you use `bps.SubTopic` in your code. Just give it a dummy file.

File: subscribe.go (hunk 1 above)
Comment (mxmCherry) [reply subscribe.go:196]: Maybe I'll get to it later ^^

**Pull request merged**